pkg/client/didexchange/client.go

```diff
@@ -119,35 +119,136 @@ func (c *Client) RemoveConnection(id string) error {
 }
 
 // startServiceEventListener listens to action and message events from DID Exchange service.
-func (c *Client) startServiceEventListener() error {
-	// register the action event channel
+func (c *Client) startServiceEventListener() {
+	// listen for action events
+	go func() {
```

Comment:
Did you desire to have two goroutines?

(note, you can alternatively use a for { select {} } within a single goroutine).

Reply:
No, missed to refactor that. Thanks for pointing out.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/e52e576769b0be364b0393262e689ca6f6dc369f..d58b6228a75eddb0dd13d89c07ec443bf5b490ec).